cockatrice/src/messagelogwidget.cpp

```diff
@@ -558,18 +558,21 @@ void MessageLogWidget::logReverseTurn(Player *player, bool reversed)
                                 .arg(reversed ? tr("reversed") : tr("normal")));
 }
 
-void MessageLogWidget::logRollDie(Player *player, int sides, int roll)
+void MessageLogWidget::logRollDie(Player *player, int sides, const QString &rolls)
 {
-    if (sides == 2) {
-        QString coinOptions[2] = {tr("Heads") + " (1)", tr("Tails") + " (2)"};
-        appendHtmlServerMessage(tr("%1 flipped a coin. It landed as %2.")
-                                    .arg(sanitizeHtml(player->getName()))
-                                    .arg("<font class=\"blue\">" + coinOptions[roll - 1] + "</font>"));
-    } else {
-        appendHtmlServerMessage(tr("%1 rolls a %2 with a %3-sided die.")
-                                    .arg(sanitizeHtml(player->getName()))
-                                    .arg("<font class=\"blue\">" + QString::number(roll) + "</font>")
-                                    .arg("<font class=\"blue\">" + QString::number(sides) + "</font>"));
+    for (const auto &rollString : rolls.split(",")) {
```

Comment:
I think it'd be a lot cleaner if we were to compound these messages, for one die keep it the same but for multiple I think something like "player rolls 3 20 sided dice: 11, 14, 20" the list would be sorted.

Reply:
I'm not sure I like the idea of sorting the list, but I'm okay with compounding them

Reply:
sorting would help so much with parsing the results though, if the order of the dice were important then using multiple separate rolls is still an option